elastic/apm-server
# Add initial support for beats central management

## Motivation/summary

Registers a reloadable object in Central Management so that when APM Server runs under Elastic Agent, it can receive and apply configuration from it.

In practice APM Server will only expect configuration once. This is enough for the standalone mode.

## Checklist


I have considered changes for:

## How to test these changes

Pull in https://github.com/elastic/beats/pull/21225 and build the Agent binary
Add an `apm` input to `elastic-agent.yml` with a well known config, eg:

```
inputs:
  - type: apm
    rum.enabled: true
    ...
``` 
Check that apm-server is running with RUM enabled

## Related issues

Required for https://github.com/elastic/apm-server/issues/4004
